perf-tool/src/server.cpp

```diff
@@ -158,24 +159,56 @@ void Server::handleServer()
             }
         }
 
-        // Receiving and sending messages from/to clients
+        /* Receiving messages to clients */
         for (int i = 0; i < activeNetworkClients; i++)
         {
-            bzero(buffer, 512);
-            command = networkClients[i]->handlePoll(buffer);
+            command = networkClients[i]->handlePoll();
             if (!command.empty())
             {
                 handleClientCommand(command, "Client");
             }
         }
+
+        /* Checks if it is the time for any delayed command to be fired */
+        if (!delayedCommands.empty()) {
+            auto currentClockTime = std::chrono::system_clock::now();
+            std::time_t currentTime = std::chrono::system_clock::to_time_t(currentClockTime);
+
+            for (int i = 0; i < delayedCommands.size(); i++)
+            {
+                if (delayedCommands[i].delayTill <= currentTime) {
+                    agentCommand(delayedCommands[i].command);
+                    delayedCommands.erase(delayedCommands.begin()+i);
+                } else{
+                    break;
+                }
+            }
+        }
     }
 }
 
 void Server::execCommand(json command)
 {
-    //if (!command["delay"].is_null())
-    //    sleep(command["delay"]);
-    if ((command["functionality"].get<std::string>()).compare("perf"))
+    if (command.find("delay") != command.end() && command["delay"].get<int>() > 0) {
+        auto currentClockTime = std::chrono::system_clock::now();
+        std::time_t currentTime = std::chrono::system_clock::to_time_t(currentClockTime);
+        std::time_t delay_till = currentTime + command["delay"].get<int>();
+
+        delayed_command_t delayedCommand;
+        delayedCommand.command = command;
+        delayedCommand.delayTill = delay_till;
+        delayedCommands.push_back(delayedCommand);
```

Comment:
If you create a constructor for `delayed_command_t` you could use:
`delayedCommands.emplace_back(command, delay_till)`